docs/ai/conceptual/prompt-engineering-in-dot-net.md

```diff
@@ -0,0 +1,190 @@
+
+To make sure the model uses the categories you want, you could append supporting content to specify your categories and adjust your instruction accordingly. You could add a line of supporting content below the instruction and then change the instruction so it refers to the supporting content:
+
+```csharp
+prompt = @$"Instructions: Summarize US Presidential accomplishments, grouped by category.
+Categories: Domestic Policy, US Economy, Foreign Affairs, Space Exploration.
+Accomplishments: George Washington
+First president of the United States.
+First president to have been a military veteran.
+First president to be elected to a second term in office.
+First president to receive votes from every presidential elector in an election.
+First president to fill the entire body of the United States federal judges; including the Supreme Court.
+First president to be declared an honorary citizen of a foreign country, and an honorary citizen of France.
+John Adams ..."; //Text truncated
+```
+
+## Examples
+
+This section explains the use of examples in .NET prompt engineering.
+
+An example is text that shows the model how to respond by providing sample user input and model output. The model uses examples to infer what to include in completions. Examples can come either before or after the instructions in an engineered prompt, but the two shouldn't be interspersed.
+
+Like a normal GPT interaction, an example starts with a prompt. The example can include a completion but it's not required. A completion in an example doesn't have to include the verbatim response&mdash;it might just contain a formatted word, the first bullet in an unordered list, or something similar to indicate how each completion should start.  
+
+Examples are classified as zero-shot learning or few-shot learning based on whether they contain verbatim completions.
+
+- **Zero-shot learning** examples include a prompt with no verbatim completion. Because they don't include verbatim completions, zero-shot prompts test a model's responses without giving it example data output. (Zero-shot prompts can have  completions that include cues, such as indicating the model should output an ordered list by including **"1."** as the completion.)
+- **Few-shot learning** examples include several pairs of prompts with verbatim completions. Few-shot learning can change the model's behavior by adding to its existing knowledge.
+
+## Cues
+
+A cue is a line of text that conveys the desired structure or format of output. Like an instruction, a cue isn't processed by the model as if it were user input. Like an example, a cue shows the model what you want instead of telling it what to do. You can add as many cues as you want, so you can iterate to get the result you want. Cues are used with an instruction or an example and should be at the end of the prompt.
```

Comment:
```suggestion
A cue is text that conveys the desired structure or format of output. Like an instruction, a cue isn't processed by the model as if it were user input. Like an example, a cue shows the model what you want instead of telling it what to do. You can add as many cues as you want, so you can iterate to get the result you want. Cues are used with an instruction or an example and should be at the end of the prompt.
```

Since the example shows a multi-line cue I figured this could be removed.